### ISSUE TITLE
Error: “Please ensure that you are using the latest version of the official game releases. Your score will not be submitted.”

### ISSUE THREAD

**Issue by Epholy** [ppy/osu#30751]
### Type

Other

### Bug description

This issue started occurring last month; prior to that, I was able to upload my scores without any problems.
I've tried various solutions suggested by other users, such as restarting my system and reinstalling osu!lazer, but none of these have resolved the issue.

I noticed an error message in the "auth.log" file:
```text
2024-11-18 07:15:55 [verbose]: Downloading update to C:\Users\Epholy\AppData\Local\Temp\tmphhlyyc.tmp...
2024-11-18 07:15:59 [verbose]: Unloading module...
2024-11-18 07:15:59 [verbose]: Module unloaded successfully.
2024-11-18 07:16:00 [verbose]: Failed to update: System.UnauthorizedAccessException: Access to the path is denied.
2024-11-18 07:16:00 [verbose]: at System.IO.FileSystem.MoveFile(String sourceFullPath, String destFullPath, Boolean overwrite)
```
It's strange that the log doesn't provide any details about the exact path that cannot be reached. And I've verified that I have read/write permissions for the Temp folder. So, what could be causing this problem?

### Screenshots or videos

![image](https://github.com/user-attachments/assets/b5b13873-1ddd-41c7-a210-3649e248ae15)

### Version

2024.1115.3

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17796875/compressed-logs.zip)

**Comment by bdach** [comment]
I've seen this a few times now, various people apparently not having their `%localappdata%/Temp` writable _to the local user_, for whatever reason. No idea what causes this.

Do you have some other software that could be running interference? Persistent file scanners? Antivirus software?

**Comment by smoogipoo** [comment]
In addition to the above, I see you've changed osu!'s folder:
```
2024-11-18 07:15:53 [verbose]: Attempting to use custom storage location D:\Program Files\osu-lazer
2024-11-18 07:15:53 [verbose]: Storage successfully changed to D:\Program Files\osu-lazer.
```
Check you have permissions to write to that folder.

**Comment by Epholy** [comment]
> I've seen this a few times now, various people apparently not having their `%localappdata%/Temp` writable _to the local user_, for whatever reason. No idea what causes this.
> 
> Do you have some other software that could be running interference? Persistent file scanners? Antivirus software?

I turned off my antivirus software, but it still doesn't work.

**Comment by Epholy** [comment]
> In addition to the above, I see you've changed osu!'s folder:
> 
> ```
> 2024-11-18 07:15:53 [verbose]: Attempting to use custom storage location D:\Program Files\osu-lazer
> 2024-11-18 07:15:53 [verbose]: Storage successfully changed to D:\Program Files\osu-lazer.
> ```
> 
> Check you have permissions to write to that folder.
> 
> _Additionally_, check if you have permissions to write to `%APPDATA%/osu-lazer`.

Yes, I do have permissions to write to these folders.
![image](https://github.com/user-attachments/assets/d84edb84-6ee1-4bb3-8c9c-a722d3e32167)
![image](https://github.com/user-attachments/assets/87e3160b-5ffa-4f4e-9322-1d0be0667fa3)
"曹Epholy" is my local account. "完全控制" means "Full control".

**Comment by necocat0918** [comment]
How about try and open lazer in administrator mode

**Comment by frenzibyte** [comment]
@necocat0918 please refrain from posting bad advices.

**Comment by People-11** [comment]
I have a possibility: did you change your account after installing osu? (e.g. switching between local and online accounts), or have you adjusted the settings regarding UAC (User Account Control)?

I came up with this idea because, for reasons of personal convenience, I've been using a global Administrator account since my system was installed, which means that for all the software I run use the Administrator privileges for file operations, including creating folders. But one day when I fell back to my normal account for debugging purposes, I found that most of the software couldn't read the files they had previously created unless I right-clicked ‘Run as administrator’.

**Comment by smoogipoo** [comment]
@Epholy 

My bad...
- Check permissions to `%LOCALAPPDATA%/osu`.
- There should already be an `AuthNative.dll` file in there - check permissions to _it_. As above if you were running with a global administrator account then it might have been created under the administrator user account. Try deleting the existing file.

**Comment by Epholy** [comment]
Oh, I think I've figured it out. As @People-11 pointed out, the ```D:\Program Files``` folder not only had osu!lazer installed but also other software installed as an administrator, which caused the owner of ```D:\Program Files``` to be ```Administrators``` instead of my personal account (```曹Epholy```). When I moved the game folder from ```D:\Program Files``` (owned by ```Administrators```) to ```E:\TEMP``` (owned by ```曹Epholy```), the modules could be updated without any problems, and scores could be submitted successfully.

Thanks to everyone's help💕, I think this issue can be closed.
